enzyme/Enzyme/GradientUtils.h

```diff
@@ -1010,37 +1010,42 @@ class GradientUtils : public CacheUtility {
     }
 
     ptr = invertPointerM(ptr, BuilderM);
-    if (!isOriginalBlock(*BuilderM.GetInsertBlock()))
-      ptr = lookupM(ptr, BuilderM);
 
-    if (!mask) {
-      auto ts = BuilderM.CreateStore(newval, ptr);
-      if (align)
+    auto rule = [&](Value *ptr, Value *newval) {
+      if (!isOriginalBlock(*BuilderM.GetInsertBlock()))
+        ptr = lookupM(ptr, BuilderM);
+
```

Comment:
The lookup should be hoisted so it's not done multiple times.

Moreover, (with the exception of the mask) should this not be okay to still perform the regular (now extra wide) store?

Also need to ponder if we need a helper alignment for shadow function

Reply:
An extra wide store is not possible since we get n different pointers (which may not point to the same region in memory).

Reply:
Ah misunderstood this as setDiffe rather than setPtrDiffe, you're absolutely right.